src/java.base/share/classes/jdk/internal/foreign/abi/ProgrammableInvoker.java

```diff
@@ -212,7 +212,7 @@ private MethodHandle specialize(MethodHandle leafHandle) {
 
         for (int i = 0; i < highLevelType.parameterCount(); i++) {
             List<Binding> bindings = callingSequence.argumentBindings(i);
-            argInsertPos += bindings.stream().filter(Binding.VMStore.class::isInstance).count() + 1;
+            argInsertPos += Math.toIntExact(bindings.stream().filter(Binding.VMStore.class::isInstance).count()) + 1;
```

Comment:
My gut feeling in this case would be that it's a bit strange to use `Math.toIntExact` to do a safe cast when you don't do `Math.addExact` to ensure that the result of the addition will not overflow. I wonder if a simple cast wouldn't be more appropriate - unless you really think that you might have more than Integer.MAX_VALUE bindings (which I doubt :-) ). But that's just my feeling...

Reply:
Yeah, that's true. I don't really expect this to ever overflow, that would create issues in other cases as well (since the limit on argument count is 255 for instance).

Okay, I'll switch this to a regular cast. (It doesn't matter to much I think because I'm also planning to remove this code: https://github.com/openjdk/jdk/pull/8685)

Reply:
Fixed.